src/SourceGenerators/Uno.UI.SourceGenerators.Tests/XamlCodeGeneratorHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿#if NET6_0_OR_GREATER
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Uno.UI.SourceGenerators.XamlGenerator;
+
+namespace Uno.UI.SourceGenerators.Tests;
+
+internal static class XamlCodeGeneratorHelper
+{
+	public static async Task<IReadOnlyCollection<Diagnostic>> RunXamlCodeGeneratorForFileAsync(
```

Comment:
Thank you. Haven't used this package before, I'll look into it and add it.

Reply:
I think it should very much simplify things. You can see https://github.com/dotnet/roslyn-analyzers/tree/main/src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests for an example of its usage.

Reply:
For the future - Switching to Microsoft.CodeAnalysis.SourceGenerators.Testing will require additional steps for the support of testing `IIncrementalGenerator`, but the implementation is presented here - https://github.com/dotnet/roslyn-sdk/issues/940

Reply:
@[Youssef1313](https://github.com/Youssef1313) I have already spent quite a lot of time looking for how to set global and per-file values for `AnalyzerConfigOptions` via `SourceGeneratorTest.TestState`. Perhaps you know?

Reply:
@HavenDV Are you trying to set `build_[metadata|property].*` options?

See https://github.com/dotnet/roslyn-analyzers/blob/5b0b341f5202a855179864a5b7cad0aba13d5582/src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/ResxGeneratorTests.cs#L430-L450

Reply:
Thank you, I did not notice this, but I suspected that there was some kind of similar format for writing through text

Reply:
Since Snapshot testing is not suitable for us, we will run tests with `TestBehaviors.SkipGeneratedSourcesCheck`. But I don't see any way to see this generated code after `test.RunAsync`. Do you know a way to do this?

Reply:
I'm not sure what you mean. `TestBehaviors.SkipGeneratedSourcesCheck` seems the complete opposite of seeing the generated code. If you want to see the generated code then you shouldn't do have `SkipGeneratedSourcesCheck`

Reply:
If we do not use this flag, then we will have to change the generated code in the tests constantly with any change in the generator (via TestState.GeneratedSources). I asked Jerome about this, and he replied that such tests are not suitable for the project at the moment. We only need to check for diagnostics.
But at the same time, the ability to look at the generated code is needed when debugging. In the current solution, I use saving to an ignored directory, so it is possible to quickly see the result.

Reply:
Do you mean validating diagnostics without validating the generated sources? In that case I think you can use `ExpectedDiagnostics`

Reply:
> But at the same time, the ability to look at the generated code is needed when debugging.

If anyone wanted to see the generated code for debugging purposes, then temporarily remove `SkipGeneratedSourcesCheck`, and add it back after fixing the issue being debugged.

Reply:
Although it will be a bit inconvenient compared to IDE access and code highlighting, I think it's acceptable. I originally asked about programmatic access to this generated code, perhaps through some kind of hidden state.

Reply:
I think it could be kept simple for now. In future, if the need to see the generated code without removing `SkipGeneratedSourcesCheck` arises, I can take a look into ways to do that.

Reply:
I added the use of the main simplification(ReferenceAssemblies) Microsoft.CodeAnalysis.SourceGenerators.Testing. As for the rest, there are two minuses:
1. Problems with access to GeneratedTrees. https://github.com/dotnet/roslyn-sdk/issues/980
2. Problems with the fact that IIncrementalGenerator is not supported at the moment (and it is not clear when it will be).

In general, I believe that at the moment a complete transition to Microsoft.CodeAnalysis.SourceGenerators.Testing will create more problems and inconvenience, so it's worth continuing with the current implementation for now and looking at it in the future.

Reply:
Thanks for the research. Let's follow this topic when more features are getting included, then we'll transition.